src/mumble/AudioInput.cpp

```diff
@@ -1080,8 +1080,10 @@ void AudioInput::encodeAudioFrame(AudioChunk chunk) {
 	bIsSpeech = bIsSpeech || (Global::get().iPushToTalk > 0);
 
 	ClientUser *p = ClientUser::get(Global::get().uiSession);
+	bool bTalkingWhenMuted = false;
 	if (Global::get().s.bMute || ((Global::get().s.lmLoopMode != Settings::Local) && p && (p->bMute || p->bSuppress))
 		|| Global::get().bPushToMute || (voiceTargetID < 0)) {
+		bTalkingWhenMuted = bIsSpeech && !(p && (p->bDeaf || p->bSelfDeaf));
```

Comment:
1. What you want to check here is `Global::get().s.bDeaf` - that's where the local user's deaf status is (somewhat) reliably stored.
2. I think semantically the check for whether the user is deafened should not be done here as the user might still be talking while being muted. _But_ in that case we don't need to play the cue (though if we did there would probably be no difference since the user won't hear it anyways ^^). In short: I would check for this further down when you actually decide whether or not to play the cue.

Reply:
Ack, implemented both. Thank you!